otherlibs/stdune/src/fpath.ml

```diff
@@ -115,6 +115,21 @@ let win32_unlink fn =
 
 let unlink = if Stdlib.Sys.win32 then win32_unlink else Unix.unlink
 
+type unlink_status =
+  | Success
+  | Is_a_directory
+  | Other_error
```

Comment:
Our think the convention we follow elsewhere is use `Error of Unix_error.Detailed.t` to preserve information, e.g. for logging.

Reply:
Done (but we also ignore other kinds of errors so the argument has type `exn`)